### ISSUE TITLE
False positive "Roll back dependency"

### ISSUE THREAD

**Issue by abraham** [renovatebot/renovate#2148]
**What Renovate type are you using?**
Renovate GitHub App

**Describe the bug**
1) I published v0.15.0 of [@nutmeg/cli](https://www.npmjs.com/package/@nutmeg/cli) to NPM.
2) I updated that dep in a project manually https://github.com/abraham/twitter-status/pull/52.
3) Renovate detects the change to the project but doesn't know about the package update.
3) Renovate opened a rollback PR https://github.com/abraham/twitter-status/pull/53.

**To Reproduce**
1) Publish a package update to NPM.
2) Update that package in a Renovate managed project before renovate knows about the update.

**Expected behavior**
I would expect the Renovate package version cache to be more current and know about about updates so as to not incorrectly roll them back.

**Comment by rarkins** [comment]
Unfortunately we're at the mercy of npm's own CDN/cache here. We only cache npm results *per repository*, meaning that this repository would have checked for @nutmeg/cli fresh at the time of run and been told that the latest was 0.14.0. When we are notified of an updated package via our npmjs listener we currently add a 1 minute wait before enqueuing any repositories, but it's not always enough. In this case it would have been a GitHub webhook notifying us of the `package.json` update that triggered an immediately renovation.

FYI Renovate's servers are located in the AWS Oregon cluster so in theory there's no geographic explanation - sometimes npm's propagation is just slow.

**Comment by abraham** [comment]
@rarkins I'm now seeing it where Renovate will open an update PR, I'll merge it and then Renovate will open a rollback PR. So it's not even consistent per repository.

https://github.com/abraham/github-repository/pull/46
https://github.com/abraham/github-repository/pull/47

**Comment by rarkins** [comment]
I just checked the logs from the last couple of days and rollback PRs all but one are scoped modules, which makes me 99% sure this is a npm cache/CDN problem. I guess one mediating solution would be to add a configuration option "rollBacks" defaulting to `true`, that then could be disabled either globally or for certain packages (e.g. starting with `@` or `@nutmet/`) via configuration.

**Comment by rarkins** [comment]
This problem has not gone away. It's still essentially an exclusive scoped problem, which makes it 99% certain to be npm backend/cache problem and not a Renovate one, otherwise we'd be seeing 99% being non-scoped instead, as is the usual ratio.

**Comment by rarkins** [comment]
I think I've discovered the root cause of this. The npmjs registry is returning a 4 hour cache time for scoped packages vs 5 minutes for non-scoped. This means that old versions can still be cached/returned for up to 4 hours after publish, depending on if you get "unlucky" and your query lands on a machine that queried the same package within the last 4 hours. I have disabled caching for scoped packages until npmjs fixes this.
